Scripts/script-CommonServer.yml

```diff
@@ -1365,6 +1365,69 @@ script: |-
       return false;
   }
 
+  /**
+   * Converts all object (and array) values of a given object to human-readable strings
+   * @param {Object} obj - the object to be altered
+   */
+  function jsonToHumanReadable(obj) {
+
+    /* Helper function that rucurses an object, formatting it's values as strings */
+    var jsonToHumanReadableRecursive = function(obj, depth) {
+        var res = "";
+
+        /* Default depth is 0 */
+        if (depth === undefined) {
+            depth = 0;
+        }
+
+        /* Return strings as is */
+        if (typeof obj === 'string') {
+            return obj;
+        }
+
+        /* Recurse all object keys, formatting values as independent lines, increasing depth with each call */
+        Object.keys(obj).forEach(function(key) {
+            var value = obj[key];
+
+            /* Handle object values in a new call */
+            if (typeof value === 'object') {
+                if (obj instanceof Array) {
+                    res += jsonToHumanReadableRecursive(value, depth) + "\n";
+                } else {
+                    res += indent(depth) + key + "\n";
+                    res += jsonToHumanReadableRecursive(value, depth + 1);  // no \n here
+                }
+            } else {
+                /* Format primitives as a string */
+                var keyStr = (obj instanceof Array) ? (parseInt(key) + 1) : key;    // 1-based indexing for arrays for better readability
+                res += indent(depth) + keyStr + ": " + value + "\n";
+            }
+        });
+
+        return res;
+    }
+
+    /* Helper function that handles the padding of new lines by depth */
+    var indent = function(depth) {
+        var res = "";
+        if (undefined === depth) {
+            depth = 0;
+        }
+
+        for (var i = 0; i < depth; i++) {
+            res += "  ";
+        }
+        return res;
+    }
+
+    /* Run all object keys and invoke the recursive helper function */
+    Object.keys(obj).forEach(function(key) {
+        var value = obj[key];
+        var str = jsonToHumanReadableRecursive(value);
+        obj[key] = str;
```

Comment:
create another object and return it, don't mutate the input parameter

Reply:
And the results should be string (not object)

Reply:
As far as cloning the object, I agree!

Regarding the 2nd comment:
The recursive function should always return a string, but the wrapper function needs to return an object (containing the fields as strings) so the it could be converted to markdown.

For example, if in our case a string that represents the whole object is returned, the next function (which is resToMD), will not create the table, but instead create 1 big cell containing the entire string.
Let's look at the original issue's image (https://user-images.githubusercontent.com/3792355/31864214-5a745966-b762-11e7-8e4b-c053f7643aba.png), I've tried to keep it exactly the same, just without the stringified objects (In other words, 1 big object containing string fields), so the same markdown will be applied.

Reply:
👍